sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -51,7 +53,7 @@ case class BroadcastHashJoinExec(
     "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
 
   override def requiredChildDistribution: Seq[Distribution] = {
-    val mode = HashedRelationBroadcastMode(buildKeys)
+    val mode = HashedRelationBroadcastMode(buildBoundKeys)
```

Comment:
Why do the keys need to be bound?

Reply:
Because this `mode` is  passed to `BroadcastExchangeExec` and eventually used to create `UnsafeProjection`. Btw, this is an existing code, but renamed per [this discussion](https://github.com/apache/spark/pull/28676#discussion_r448124006).